data-pipeline/src/trace_exporter.rs

```diff
@@ -1006,15 +1200,20 @@ mod tests {
             .build()
             .unwrap();
 
-        let mut trace_chunk = vec![pb::Span {
+        let trace_chunk = vec![pb::Span {
             duration: 10,
             ..Default::default()
         }];
 
-        trace_utils::compute_top_level_span(&mut trace_chunk);
-
         let data = rmp_serde::to_vec_named(&vec![trace_chunk]).unwrap();
 
+        // Wait for the info fetcher to get the config
+        while mock_info.hits() == 0 {
+            exporter.runtime.block_on(async {
+                sleep(Duration::from_millis(100)).await;
```

Comment:
since this is in a loop, would it make sense to lower the duration?

Reply:
I tried a lower duration with the info fetcher but this sometimes failed probably due to a race condition where the loop ended because the server received the request but the response had not yet been received by the client.